hw_13/internet_Sotnikov.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import requests
+import re
+import os
+from dataclasses import dataclass
+
+@dataclass
+class GenscanOutput:
+    status: str
+    cds_list: list
+    intron_list: list
+    exon_list: list
+        
+    def __repr__(self):
+        nl = f"\n"
+        return f"Response status: {self.status}\n\nCds_list: {self.cds_list}\n\nIntron_list (number, start, end): {self.intron_list}\n\nExon_list (number, strans, start, end): {self.exon_list}"
+
+    
+def run_genscan(sequence=None, sequence_file=None, organism="Vertebrate", exon_cutoff=1.00, sequence_name=""):
+    data = {"-p": "Predicted CDS and peptides", "-o": organism, "-e": exon_cutoff, "-n": sequence_name}
+    url = "http://hollywood.mit.edu/cgi-bin/genscanw_py.cgi"
+    
+    
+    if sequence is None and sequence_file is not None:
+        
+        with open(sequence_file, "r") as dna_seq:
+            dna_strings = dna_seq.readlines()    
+        number, lim = 0, len(dna_strings)
+        while number < lim:
+            string = dna_strings[number]
+            if string.startswith(">"):
+                dna_strings.remove(string)
+                lim = len(dna_strings)
+            else:
+                number += 1
+        
+        data["-u"] = "".join(dna_strings)
+        response = requests.post(url, data=data)
+        
+    elif sequence is not None and sequence_file is None:
+        data["-s"] = sequence
+        response = requests.post(url, data=data)
+        
+    elif sequence is not None and sequence_file is not None:
+        raise Warning("You have passed both sequence and sequence_file arguments. \n\
+        Pass only on of these arguments, please.")
+
+    elif sequence is None and sequence_file is None:
+        raise ValueError("You can't post both sequence and sequence_file as None. You need to put data into one of them")
```

Comment:
Я бы немного поменял порядок, чтобы избежать повторений кода. Сначала проверки, потом действия
```suggestion
    if sequence is not None and sequence_file is not None:
        raise Warning("You have passed both sequence and sequence_file arguments. \n\
        Pass only on of these arguments, please.")

    elif sequence is None and sequence_file is None:
        raise ValueError("You can't post both sequence and sequence_file as None. You need to put data into one of them")
        
    if sequence is None and sequence_file is not None:    
        with open(sequence_file, "r") as dna_seq:
            dna_strings = dna_seq.readlines()    
        number, lim = 0, len(dna_strings)
        while number < lim:
            string = dna_strings[number]
            if string.startswith(">"):
                dna_strings.remove(string)
                lim = len(dna_strings)
            else:
                number += 1
        
        data["-u"] = "".join(dna_strings)
             
    elif sequence is not None and sequence_file is None:
        data["-s"] = sequence
        
    response = requests.post(url, data=data)
```